test/unit/utils_test.js

```diff
@@ -68,6 +68,1210 @@ describe('Utils', function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      const unrecognizedKP = StellarSdk.Keypair.random();
+      transaction.sign(unrecognizedKP);
+
+      expect(() =>
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          challenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          [this.clientKP1.publicKey()],
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        "Transaction not signed by client(s): '" +
+          this.clientKP1.publicKey() +
+          "'",
+      );
+    });
+
+    it("successfully validates server and multiple client signers in the transaction", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      const clientSigners = [this.clientKP1, this.clientKP2];
+      transaction.sign(...clientSigners);
+      const clientSignersPubKey = clientSigners.map((kp) => {
+        return kp.publicKey();
+      });
+
+      const signedChallenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          signedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          clientSignersPubKey,
+        ),
+      ).to.eql(clientSignersPubKey);
+    });
+
+    it("successfully validates server and multiple client signers, in reverse order", function() {
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        this.serverKP,
+        this.clientKP1.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+      const clientSigners = [this.clientKP1, this.clientKP2];
+      transaction.sign(...clientSigners.reverse());
+      const clientSignersPubKey = clientSigners.map((kp) => {
+        return kp.publicKey();
+      });
+
+      const signedChallenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(
+        StellarSdk.Utils.verifyChallengeTxSigners(
+          signedChallenge,
+          this.serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+          clientSignersPubKey,
+        ),
+      ).to.have.same.members(clientSignersPubKey.reverse());
```

Comment:
Because you're using `.to.have.same.members` I don't think you need to do the `.reverse()` here. The order doesn't matter in your check.